be/src/olap/rowset/segment_v2/segment_iterator.cpp

```diff
@@ -379,7 +379,9 @@ Status SegmentIterator::_get_row_ranges_from_conditions(RowRanges* condition_row
         DCHECK(_opts.col_id_to_predicates.count(cid) > 0);
         RETURN_IF_ERROR(_column_iterators[_schema.unique_id(cid)]->get_row_ranges_by_zone_map(
                 _opts.col_id_to_predicates[cid].get(),
-                _opts.col_id_to_del_predicates.count(cid) > 0
+                _opts.col_id_to_del_predicates.size() == 1 &&
```

Comment:
Add a comment here to explain these conditions. For example, why check size == 1

Reply:
ok, i add a comment